### ISSUE TITLE
[java] CloseResource false positive if Stream is passed as method parameter

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#1966]
**Affects PMD Version:** 6.16.0

**Rule:** [CloseResource](https://pmd.github.io/latest/pmd_rules_java_errorprone.html#closeresource)

**Description:**

If a IO stream is passed into a method as a method parameter and then assigned to a local variable, the rule should not be triggered. In that case, the stream has not be created locally, so the caller of the method is responsible to close it.

**Code Sample demonstrating the issue:**

``` java
import java.io.*;

public class CloseResourceFP {
    public void check(InputStream in) {
        if (in instanceof FileInputStream) {
            FileInputStream fin = (FileInputStream) in;
            doCheck(fin);
        } else if (in instanceof ByteArrayInputStream) {
            ByteArrayInputStream bin = (ByteArrayInputSream) in;
            doCheck(bin);
        } else {
            BufferedInputStream buf = new BufferedInputStream(in);
            doCheck(buf);
        }
    }

    public void dump(final Writer writer) {
        final PrintWriter printWriter = writer instanceof PrintWriter ? (PrintWriter) writer : new PrintWriter(writer);
        printWriter.println(this);
    }
}

```
